src/test/java/uk/gov/justice/hmpps/prison/util/builders/OffenderNoPayPeriodBuilder.kt

```diff
@@ -0,0 +1,31 @@
+package uk.gov.justice.hmpps.prison.util.builders
+
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderNoPayPeriod
+import uk.gov.justice.hmpps.prison.service.BadRequestException
+import uk.gov.justice.hmpps.prison.service.DataLoaderRepository
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+class OffenderNoPayPeriodBuilder(
+  private val startDate: LocalDate = LocalDate.of(2016, 11, 9),
+  private val endDate: LocalDate? = null,
+  private val reason: String = "REFUSAL",
+) {
+  fun save(
+    bookingId: Long,
+    dataLoader: DataLoaderRepository,
+  ) {
```

Comment:
Added new builders for entities where the tests were previously relying on seed data - we now create and tear down the data.